tests/servers/good.rb

```diff
@@ -333,5 +333,6 @@ def split_header_value(str)
 
 EM.run do
   EM.start_server("127.0.0.1", 9292, GoodServer)
+  EM.start_server("::1", 9293, GoodServer)
```

Comment:
So, on jruby, it's failing because I'm starting EventMachine and binding on  ::1:9293 and trying to connect to it in the test using http://[::1]:9293.  See the failure [here](https://travis-ci.org/excon/excon/jobs/47718169).  This works on MRI 1.9.3/2.0.0/2.1.5.

I couldn't find issues saying that you can't do this on jruby but I did find this:
http://stackoverflow.com/questions/15061805/connection-refused-thrift-client-on-jruby-1-7-3

Note:  I tried replacing ::1 and [::1] above to localhost and tests pass (but then ipv6 [] wrapping/unwrapping isn't being tested).  When requiring `tests/servers/good.rb` using localhost and `lsof` I see IPv6 file descriptors "listening" but again the purpose of the tests were to verify the correct usage of ::1 and [::1] for socket and non-socket usage.

I also tried "0:0:0:0:0:0:0:1" (the long form of ::1) and that also fails with `Connection refused - Connection refused (Errno::ECONNREFUSED)`